pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1192 @@
+        - Y axis : the scale is ydiv / 25 and the offset -yoffset. the
+        offset is not applied for the MATH source.
+        - X axis : the scale is sparsing / sampling_rate and the offset is -xdiv * 7. The
+        7 = 14 / 2 factor comes from the fact that there are 14 vertical grid lines and the data
+        starts from the left half of the screen.
+
+        :return: tuple of (numpy array of Y points, numpy array of X points, waveform preamble) """
+
+        def _scale_data(x):
+            if preamble["source"] == "MATH":
+                value = int.from_bytes([x], byteorder='big', signed=False) * preamble["ydiv"] / 25.
+                value -= preamble["ydiv"] * (preamble["yoffset"] + 255) / 50.
+            else:
+                value = int.from_bytes([x], byteorder='big', signed=True) * preamble["ydiv"] / 25.
+                value -= preamble["yoffset"]
+            return value
+
+        def _scale_time(t):
+            return float(Decimal(-preamble["xdiv"] * self._grid_number / 2.) +
+                         Decimal(float(t * preamble["sparsing"])) /
+                         Decimal(preamble["sampling_rate"]))
+
+        data_points = np.vectorize(_scale_data)(xdata)
+        if len(data_points) > 1:
+            time_points = np.vectorize(_scale_time)(np.arange(len(data_points)))
+        else:
+            time_points = np.array([0])
+        return data_points, time_points, preamble
+
+    def download_data(self, source, requested_points=0, sparsing=1):
+        """ Get data points from the specified source of the oscilloscope. The returned objects are
+        two np.ndarray of data and time points and a dict with the waveform preamble, that contains
+        metadata about the waveform.
+        Note.
+        :param source: measurement source. It can be "C1", "C2", "C3", "C4", "MATH.
+        :param requested_points: number of points to acquire. If 0, all available points
+        will be returned.
+        :param sparsing: interval between data points. For example if sparsing = 4, only one
+        point every 4 points is read.
+        :return: data_ndarray, time_ndarray, waveform_preamble_dict: see waveform_preamble
+        property for dict format. """
+        if not sparsing:
+            sparsing = 1
+        if requested_points is None:
+            requested_points = 0
+        self.waveform_source = _sanitize_source(source)
+        if requested_points == 1:
+            xdata, preamble = self._acquire_one_point()
+        else:
+            xdata, preamble = self._acquire_data(requested_points, sparsing)
+        preamble["transmitted_points"] = len(xdata)
+        preamble["requested_points"] = requested_points
+        preamble["sparsing"] = sparsing
+        preamble["first_point"] = 0
+        return self._process_data(xdata, preamble)
+
+    ###############
+    #   Trigger   #
+    ###############
+
+    trigger_50 = Instrument.setting(
```

Comment:
You do not set anything, so it should not be a property. A method `center_trigger()` would be better.

Reply:
Done